include/links-directory.php

```diff
@@ -89,10 +89,10 @@ public function remove_language_from_link( $url ) {
 		}
 
 		if ( ! empty( $languages ) ) {
-			$root = ( false === strpos( $url, '://' ) ) ? $this->home_relative . $this->root : preg_replace( '/^https?:\/\//', '://', $this->home . '/' . $this->root );
+			$root = ( false === strpos( $url, '://' ) ) ? $this->home_relative . $this->root : preg_replace( '#^https?://#', '://', $this->home . '/' . $this->root );
 
-			$pattern = str_replace( '/', '\/', $root );
-			$pattern = '#' . $pattern . ( $this->options['rewrite'] ? '' : 'language\/' ) . '(' . implode( '|', $languages ) . ')(\/|$)#';
+			$pattern = preg_quote( $root, '#' );
+			$pattern = '#' . $pattern . ( $this->options['rewrite'] ? '' : 'language/' ) . '(' . implode( '|', $languages ) . ')(/|$)#';
```

Comment:
Shouldn't we also escape `$this->option['rewrite']` and '$languages`, or the whole `$pattern` after line 95?

Reply:
`$this->option['rewrite']` isn't used, `$languages` can be escaped but it's only a list of slugs (`[a-z-]`) so won't cause issues.

Reply:
I don't think so. This would escape the pipes and parentheses.